### PR TITLE
Added doc comments

### PR DESCRIPTION
**Pull request by blowdart**

### CODE REVIEW

**Comment by HaoK** [comment]
@divega We want these for RC1?  So do you want me to merge these into release?

File: src/Microsoft.AspNet.Identity.EntityFramework/IdentityRole.cs [hunk 1]
```diff
@@ -7,22 +7,28 @@
...
 {
     /// <summary>
-    ///     Represents a Role entity
+    /// The default implementation of <see cref="IdentityRole{TKey}"/> which uses a string as the primary key.
     /// </summary>
     public class IdentityRole : IdentityRole<string>
     {
         /// <summary>
-        ///     Constructor
+        /// Initializes a new instance of <see cref="IdentityRole{TKey}"/>.
```
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityRole.cs]: This is the non generic version.

File: src/Microsoft.AspNet.Identity.EntityFramework/IdentityRole.cs [hunk 2]
```diff
@@ -7,22 +7,28 @@
...
+        /// The Id property is initialized to from a new GUID string value.
+        /// </remarks>
         public IdentityRole()
         {
             Id = Guid.NewGuid().ToString();
         }
 
         /// <summary>
-        ///     Constructor
+        /// Initializes a new instance of <see cref="IdentityRole{TKey}"/>.
```
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityRole.cs]: Non generic.

File: src/Microsoft.AspNet.Identity.EntityFramework/IdentityUser.cs [hunk 3]
```diff
@@ -6,108 +6,155 @@
 
 namespace Microsoft.AspNet.Identity.EntityFramework
 {
+    /// <summary>
+    /// The default implementation of <see cref="IdentityUser{TKey}"/> which uses a string as a primary key.
+    /// </summary>
     public class IdentityUser : IdentityUser<string>
     {
+        /// <summary>
+        /// Initializes a new instance of <see cref="IdentityUser{TKey}"/>.
```
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityUser.cs]: Non generic

File: src/Microsoft.AspNet.Identity.EntityFramework/IdentityUser.cs [hunk 4]
```diff
@@ -6,108 +6,155 @@
...
+        /// <remarks>
+        /// The Id property is initialized to from a new GUID string value.
+        /// </remarks>
         public IdentityUser()
         {
             Id = Guid.NewGuid().ToString();
         }
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="IdentityUser{TKey}"/>.
```
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityUser.cs]: Non generic

File: src/Microsoft.AspNet.Identity.EntityFramework/UserStore.cs [hunk 5]
```diff
@@ -50,6 +70,11 @@ public class UserStore<TUser, TRole, TContext, TKey> :
         where TKey : IEquatable<TKey>
     {
 
+        /// <summary>
+        /// Creates a new instance of <see cref="UserStore"/>.
+        /// </summary>
+        /// <param name="context">The context used to access the store.</param>
+        /// <param name="describer">The <see cref="IdentityErrorDescriber"/> to place store errors into.</param>
```
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/UserStore.cs]: Not sure this is the best description. Perhaps "used to obtain descriptions for errors returned by the store"? @haok?

File: src/Microsoft.AspNet.Identity.EntityFramework/UserStore.cs [hunk 6]
```diff
@@ -263,11 +364,11 @@ public virtual Task SetPasswordHashAsync(TUser user, string passwordHash, Cancel
         }
 
         /// <summary>
-        ///     Get the password hash for a user
+        /// Gets the password hash for a user, as an asynchronous operation.
```
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/UserStore.cs]: Does the "as an async operation" part add much on all these?

File: src/Microsoft.AspNet.Identity.EntityFramework/IdentityEntityFrameworkServices.cs [hunk 7]
```diff
@@ -11,7 +11,7 @@ namespace Microsoft.AspNet.Identity
     /// <summary>
     /// Default services
     /// </summary>
-    public class IdentityEntityFrameworkServices
+    internal class IdentityEntityFrameworkServices
```
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityEntityFrameworkServices.cs]: I don't think we should make this internal. @haok didn't we get rid of this pattern in other repos?

**Comment by divega** [comment]
Very nice! Only minor comments/questions.

File: src/Microsoft.AspNet.Identity.EntityFramework/IdentityEntityFrameworkServices.cs (hunk 7 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityEntityFrameworkServices.cs]: Hence why I asked him to make it internal, we got rid of the public pattern yes.  But we call this code in two places (with a different key type), so its useful as a helper method.  I can move it into the Builder extension class as a private instead, but the idea was to make this go away from a public API perspective

File: src/Microsoft.AspNet.Identity.EntityFramework/UserStore.cs (hunk 6 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Identity.EntityFramework/UserStore.cs]: Yeah just seems like extra verbiage.

File: src/Microsoft.AspNet.Identity.EntityFramework/IdentityEntityFrameworkServices.cs (hunk 7 above)
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityEntityFrameworkServices.cs]: Private method sounds good. I am fine with doing it in two steps.

**Comment by HaoK** [comment]
Updated

**Comment by HaoK** [comment]
379f1928a4d676d2ad14ba026f50b204ed33d518

**Pull request closed without merging**